_rules/SC1-3-1-aria-required-context-role.md

```diff
@@ -0,0 +1,198 @@
+---
+name: ARIA required context role
+description: | 
+  This rule checks that a role does not exist outside of its required context roles
+
+success_criterion:
+- 1.3.1 # Info and Relationships
+
+test_aspects: 
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any HTML or SVG element that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has an explicit [semantic role](#semantic-role) with a [WAI-ARIA required context role](https://www.w3.org/TR/wai-aria-1.1/#scope), except if the element has an implicit semantic role that is identical to its explicit semantic role.
```

Comment:
I would add/change the following to Applicability:
The rule applies to any **non empty** HTML or SVG element.
Screen readers do not read the empty attributes. If there is no content screenreaders would ignore if empty.
This means that the fail examples must be adjusted.

Reply:
@danistr, what do you mean by "non-empty"? It's not this definition, is it? https://auto-wcag.github.io/auto-wcag/pages/glossary/non-empty.html

Reply:
@annethyme We would not fail any of the failed examples because they hold no content. Screenreaders do not read the empty elements. 

That definition comes very close to what we mean. A suggestion is to call it 'with content'. For example: The rule applies to any HTML or SVG element with content.

Reply:
@danistr, sorry for taking so long to get back to you on this. 

The definitions we have tried out so far for "non-empty" has been for text only. 
What we need here is something that will work for elements. 

@kasperisager and I gave it a shot.
(And @kasperisager, please correct me if I got the wrong iteration of the many we went through).  

*Element has content available to assistive technologies:*

> The element, or any of its [descendants] in the [flat tree] that are [included in accessibility tree], has either:
> - an [accessible name] that is not only [whitespace],
> - children that are [text nodes] that are not only [whitespace], or
> - children that are `img` elements that are [included in the accessibility tree].

Reply:
@WilcoFiers, I would appreciate your opinion too on the above shot at a definition of when elements have content seen from an AT perspective.

Reply:
@danistr, we just tested in VoiceOver, and it *does* read empty elements, so we would suggest keeping it as is, actually.
@WilcoFiers, how does aXe Core handle this?

Reply:
Discussed on the Auto-WCAG call. There seems to be big differences between different AT and browsers. 
The group suggests just adding an Accessibility Support section that explains that some AT/browsers will ignore empty elements.